lightning-persister/src/lib.rs

```diff
@@ -108,39 +102,64 @@ impl FilesystemPersister {
 		util::write_to_file(path, "manager".to_string(), manager)
 	}
 
-	#[cfg(test)]
-	fn load_channel_data<Keys: KeysInterface>(&self, keys: &Keys) ->
-		Result<HashMap<OutPoint, ChannelMonitor<Keys::Signer>>, ChannelMonitorUpdateErr> {
-			if let Err(_) = fs::create_dir_all(self.path_to_monitor_data()) {
-				return Err(ChannelMonitorUpdateErr::PermanentFailure);
+	/// Read `ChannelMonitor`s from disk.
+	pub fn read_channelmonitors<Signer: Sign, K: Deref> (
+		&self, keys_manager: K
+	) -> Result<HashMap<OutPoint, (BlockHash, ChannelMonitor<Signer>)>, std::io::Error>
+		where K::Target: KeysInterface<Signer=Signer> + Sized
+	{
+		let path = self.path_to_monitor_data();
+		if !Path::new(&path).exists() {
+			return Ok(HashMap::new());
```

Comment:
Hmm, isn't it more likely its just the first startup and thus there's no monitors?

Reply:
I was thinking that the caller would handle creating the directory. But I guess that `FilesystemPersister` creates it on demand, so probably ok then. Wasn't sure if we want to give the user insight into this (e.g., they instantiate a persister expecting to find data but don't because the path was wrong).